Source/fsSerializer.cs

```diff
@@ -474,9 +477,61 @@ internal class fsLazyCycleDefinitionWriter {
 
             var result = InternalSerialize_1_ProcessCycles(storageType, instance, out data);
             Invoke_OnAfterSerialize(processors, storageType, instance, ref data);
+
+            //remove default values serialization in the end?
+            if (!fsConfig.SerializeDefaultValues){
+                return RemoveDefaultValues(storageType, instance, ref data);
+            }
+
             return result;
         }
 
+        ///Clean up defaults. Is there a better way to do this?
```

Comment:
I think this is a good approach for removing default values, but I'd like to centralize the logic to either this file or to the printer. Right now it's split.

The printer can output either pretty or compressed JSON, so it probably makes sense to keep it in fsSerializer so it's only implemented once. Instead of setting key/values with default values to null, can we just remove them entirely? That way the printer is completely ignored that default values ever existed.